sycl/include/sycl/detail/vector_convert.hpp

```diff
@@ -586,11 +533,17 @@ NativeToT convertImpl(NativeFromT Value) {
   else if constexpr (is_float_to_uint<FromT, ToT>::value)
     return ConvertFToU<NativeFromT, NativeToT, VecSize, ElemTy, RoundingMode>(
         Value);
-  else if constexpr (is_sint_to_uint<FromT, ToT>::value)
-    return SatConvertSToU<NativeFromT, NativeToT, VecSize, ElemTy>(Value);
   else {
-    static_assert(is_uint_to_sint<FromT, ToT>::value);
-    return SatConvertUToS<NativeFromT, NativeToT, VecSize, ElemTy>(Value);
+    static_assert(is_sint_to_from_uint<FromT, ToT>::value);
```

Comment:
nit: maybe add small message here?

Reply:
Added in ba81a3bf4af09b9a9a92d1d4074cd06c689f15cd